### PR TITLE
Issue proper warning trying to access calldata variables in inline assembly

### LINKED ISSUE #2555: Show proper error if using calldata variables in inline assembly
Part of #1803.

### PR DESCRIPTION
**Pull request by axic**
Fixes #2555.

### CODE REVIEW

**Comment by axic** [comment]
Help welcome in changing the text to a useful message.

File: libsolidity/analysis/TypeChecker.cpp [hunk 1]
```diff
@@ -723,7 +723,10 @@ bool TypeChecker::visit(InlineAssembly const& _inlineAssembly)
 			}
 			else if (var->type()->sizeOnStack() != 1)
 			{
-				m_errorReporter.typeError(_identifier.location, "Only types that use one stack slot are supported.");
+				if (var->type()->dataStoredIn(DataLocation::CallData))
+					m_errorReporter.typeError(_identifier.location, "Call data elements cannot be accessed directly. Copy to a local variable first or use \"calldataload\" or \"calldatacopy\" directly.");
```
Comment (chriseth) [inline libsolidity/analysis/TypeChecker.cpp]: `directly` -> `with manually determined offsets and sizes.`?

Comment (axic) [reply libsolidity/analysis/TypeChecker.cpp]: You mean `Call data elements cannot be accessed with manually determined offsets and sizes.`?

Comment (chriseth) [reply libsolidity/analysis/TypeChecker.cpp]: `Call data elements cannot be accessed directly. Copy to a local variable first or use \"calldataload\" or \"calldatacopy\" with manually determined offsets and sizes.`

Comment (axic) [reply libsolidity/analysis/TypeChecker.cpp]: Updated.

**Review by chriseth — APPROVED** [review]

**Pull request merged**